File: java/1 - pilha/Conversor.java
public class Conversor {
  public static void main(String args[]) {
    int numero = Integer.parseInt(args[0]);
    Pilha p = new Pilha(numero);
    int resto;

    while (numero != 0) {
      resto = numero % 2; // pega o resto da divisão deste numero por 2 (vale 0 ou 1)
      p.push(resto);
      numero = numero / 2;
    }

    while (!p.isEmpty()) {
      resto = p.pop();
      System.out.print(resto);
    }
    System.out.println("\nFim do programa");
  }
}

File: c/4 - pilha dinamica/conversor.c
#include <stdio.h>
#include "pilhaDinamica.c"

int main(int argc, char *argv[])
{

  int valor, resto;
  struct Pilha p;
  create(&p);

  printf("Digite um valor ai velhinho:");
  scanf("%d", &valor);

  while (valor != 0)
  {
    resto = valor % 2;
    push(&p, resto);
    valor = valor / 2;
  }

  while (!isEmpty(p))
  {
    resto = pop(&p);
    printf("%d", resto);
  }
  printf("\n\nfim do programa.");

  return 0;
}